lib/jar.ts

```diff
@@ -28,47 +28,90 @@ interface MavenPackageInfo {
   v: string; // version
 }
 
-export async function getTempPomPathForJar(
-  root: string,
-  jarName: string,
-): Promise<string> {
-  const jarPath = path.resolve(root, jarName);
+function getSha1(buf: Buffer) {
+  return crypto
+    .createHash(ALGORITHM)
+    .update(buf)
+    .digest(DIGEST);
+}
 
+async function getMavenDependency(
+  targetPath: string,
+): Promise<MavenDependency> {
   try {
-    if (!fs.existsSync(jarPath)) {
-      throw new Error('Unable to find jar at ' + jarPath);
-    }
+    const contents = fs.readFileSync(targetPath);
+    const sha1 = getSha1(contents);
+    const { g, a, v } = await getMavenPackageInfo(sha1);
+    return {
+      groupId: g,
+      artifactId: a,
+      version: v,
+    };
+  } catch (err) {
+    throw new Error('Failed to get maven dependency. ' + err.message);
+  }
+}
 
-    const fileContents = fs.readFileSync(jarPath);
+async function getDependencies(paths: string[]): Promise<MavenDependency[]> {
+  let dependencies: MavenDependency[] = [];
+  for (const p of paths) {
+    const dependency = await getMavenDependency(p);
+    dependencies.push(dependency);
+  }
+  return dependencies;
+}
 
-    const sha1 = crypto
-      .createHash(ALGORITHM)
-      .update(fileContents)
-      .digest(DIGEST);
+export async function createPomForJar(
+  root: string,
+  targetFile?: string,
```

Comment:
What case would the targetFile not be provided?

Reply:
when you just run 'snyk test --all-jars' from the directory with jars in

Reply:
wouldn't that go to the createPomForJar`s` method?

Reply:
ah yea sorry - i'll take out of this func, tahnks